query-planner/tests/cucumber.rs

```diff
@@ -0,0 +1,102 @@
+use cucumber::cucumber;
+use std::fmt;
+
+type Result<T> = std::result::Result<T, QueryPlanError>;
+
+// stub out a custom QueryPlanError
+#[derive(Debug, Clone)]
+struct QueryPlanError;
+impl fmt::Display for QueryPlanError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "error in building query plan")
+    }
+}
+
+struct Schema {}
+fn parse_schema(_schema: &str) -> Result<Schema> {
+    Ok(Schema {})
+}
+
+struct Query {}
+fn parse_query(_query: &str) -> Result<Query> {
+    Ok(Query {})
+}
+#[derive(Debug)]
+struct QueryPlan {
+    foo: String,
+}
+impl std::cmp::PartialEq for QueryPlan {
+    fn eq(&self, other: &QueryPlan) -> bool {
+        self.foo == other.foo
+    }
+}
+
+fn build_query_plan(_schema: &Schema, _query: &Query) -> Result<QueryPlan> {
+    Ok(QueryPlan {
+        foo: "okay then".to_string(),
+    })
+}
+
+// // only used by tests for now
+fn build_query_plan_from_str(schema: &str, query: &str) -> QueryPlan {
+    let schema = parse_schema(schema).expect("failed parsing schema");
+    let query = parse_query(query).expect("failed parsing query");
+    build_query_plan(&schema, &query).expect("failed building QueryPlan")
+}
+
+pub struct QueryPlannerTestContext {
+    // You can use this struct for mutable context in scenarios.
+    query: Option<String>,
+}
+impl cucumber::World for QueryPlannerTestContext {}
+impl std::default::Default for QueryPlannerTestContext {
+    fn default() -> QueryPlannerTestContext {
+        // This function is called every time a new scenario is started
+        QueryPlannerTestContext { query: None }
+    }
+}
+
+mod query_planner_tests {
```

Comment:
Is this `mod` necessary for cucumber? This code is already in an integration test module named `cucumber`.

Reply:
I think so, yes 🙃